functions/slack/SlackKnowledgeGraphSearch/Function.cs

```diff
@@ -0,0 +1,175 @@
+            string slackSigningSecret = context.Configuration["SLACK_SECRET"];
+
+            services.AddSingleton(new KgsearchService(new BaseClientService.Initializer
+            {
+                ApiKey = kgApiKey
+            }));
+            services.AddSingleton(new SlackRequestVerifier(slackSigningSecret, SlackTimestampTolerance));
+            base.ConfigureServices(context, services);
+        }
+    }
+    // [END functions_slack_setup]
+
+    public class Function : IHttpFunction
+    {
+        private readonly ILogger _logger;
+        private readonly KgsearchService _kgService;
+        private readonly SlackRequestVerifier _verifier;
+
+        public Function(ILogger<Function> logger, KgsearchService kgService, SlackRequestVerifier verifier) =>
+            (_logger, _kgService, _verifier) = (logger, kgService, verifier);
+
+        // [START functions_slack_search]
+        public async Task HandleAsync(HttpContext context)
+        {
+            var request = context.Request;
+            var response = context.Response;
+
+            // Validate request
+            if (request.Method != "POST")
+            {
+                _logger.LogWarning("Unexpected request method '{method}'", request.Method);
+                response.StatusCode = (int) HttpStatusCode.MethodNotAllowed;
+                return;
+            }
+
+            if (!request.HasFormContentType)
+            {
+                _logger.LogWarning("Unexpected content type '{contentType}'", request.ContentType);
+                response.StatusCode = (int) HttpStatusCode.BadRequest;
+                return;
+            }
+
+            // We need to read the request body twice: once to validate the signature,
+            // and once to read the form content. We copy it into a memory stream,
+            // so that we can rewind it after reading.
+            var bodyCopy = new MemoryStream();
+            await request.Body.CopyToAsync(bodyCopy);
+            request.Body = bodyCopy;
+            bodyCopy.Position = 0;
+
+            if (!_verifier.VerifyRequest(request, bodyCopy.ToArray()))
+            {
+                _logger.LogWarning("Slack request verification failed");
+                response.StatusCode = (int) HttpStatusCode.Unauthorized;
+                return;
+            }
+
+            var form = await request.ReadFormAsync().ConfigureAwait(false);
+            if (!form.TryGetValue("text", out var query))
+            {
+                _logger.LogWarning("Slack request form did not contain a text element");
+                response.StatusCode = (int) HttpStatusCode.BadRequest;
+                return;
+            }
+
+            var kgResponse = await SearchKnowledgeGraphAsync(query);
+            string formattedResponse = FormatSlackMessage(kgResponse, query);
+            response.ContentType = "application/json";
+            await response.WriteAsync(formattedResponse);
```

Comment:
ConfigureAwait missing here as well

Reply:
Left out - I've just removed the one between these two comments.